crates/noirc_evaluator/src/ssa/optimizations.rs

```diff
@@ -98,6 +99,49 @@ fn evaluate_intrinsic(
             }
             unreachable!();
         }
+        builtin::Opcode::ToRadix(endian) => {
+            let mut element = args[0].to_le_bytes().to_vec();
+            if endian == Endian::Big {
+                element = args[0].to_be_bytes().to_vec();
+            }
+            let byte_count = args[2] as u32;
+            let diff = if byte_count > element.len() as u32 {
+                byte_count - element.len() as u32
+            } else {
+                0
+            };
+            if endian == Endian::Little {
+                element.extend(vec![0; diff as usize])
+            } else {
+                let mut new_element = vec![0; diff as usize];
```

Comment:
I think it is easier to do it with little_endian only and just reversing the result if big_endian. Since we iterate on the result after, reversing the iterator should be almost zero cost.